### PR TITLE
*: add mydumper

### PR DESCRIPTION
**Pull request by CaitinChen**
via https://github.com/pingcap/docs/pull/644
@lilin90 @kennytm PTAL. Thanks!

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -656,7 +656,7 @@ TiDB 设计的目标就是针对 MySQL 单台容量限制而被迫做的分库
 
 #### 3.6.1 TiDB 主要备份方式？
 
-目前 TiDB 主要依赖 mydumper 逻辑导出进行备份，具体可以参考 [https://github.com/maxbube/mydumper](https://github.com/maxbube/mydumper)，虽然 TiDB 也支持使用 MySQL 官方的 mysqldump 工具来进行数据的备份恢复工作，但相比于 mydumper/loader，性能会慢很多，大量数据的备份恢复会花费很多时间，这里我们并不推荐。
+目前，推荐的备份方式是使用 [PingCAP fork of mydumper](tools/mydumper.md)。尽管 TiDB 也支持使用 MySQL 官方工具 `mysqldump` 进行数据备份、恢复，但其性能低于 [`mydumper`](tools/mydumper.md)/[`loader`](tools/loader.md)，并且该工具备份、恢复大量数量时，要耗费更多时间。
```
Comment (kennytm) [inline FAQ.md]: "PingCAP fork of mydumper" can also be translated to Chinese 「PingCAP 的 mydumper 分支」。

File: FAQ.md [hunk 2]
```diff
@@ -666,14 +666,14 @@ loader的 -t 参数可以根据 TiKV 的实例个数以及负载进行评估调
 
 ### 4.1 全量数据导出导入
 
-#### 4.1.1 Mydumper 工具
+#### 4.1.1 Mydumper
 
-具体可以参考 [https://github.com/maxbube/mydumper](https://github.com/maxbube/mydumper)。
+参见 [mydumper Instructions](tools/mydumper.md)。
```
Comment (kennytm) [inline FAQ.md]: "Instructions" → "使用文档"

File: FAQ.md [hunk 3]
```diff
@@ -666,14 +666,14 @@ loader的 -t 参数可以根据 TiKV 的实例个数以及负载进行评估调
...
 
-#### 4.1.2 Loader 工具
-
-具体可以参考 [https://www.pingcap.com/docs-cn/tools/loader/](tools/loader.md)。
+#### 4.1.2 Loader
 
+参见 [Loader Instructions](tools/loader.md)。
```
Comment (kennytm) [inline FAQ.md]: "Instructions" → "使用文档"

File: op-guide/backup-restore.md [hunk 4]
```diff
@@ -38,9 +38,9 @@ cd tidb-enterprise-tools-latest-linux-amd64
 
 `mydumper` 是一个强大的数据备份工具，具体可以参考 [https://github.com/maxbube/mydumper](https://github.com/maxbube/mydumper)。
 
-我们使用 `mydumper` 从 TiDB 导出数据进行备份，然后用 `loader` 将其导入到 TiDB 里面进行恢复。
+可使用 [`mydumper`](../tools/mydumper.md) 从 TiDB 导出数据进行备份，然后用 [`loader`](../tools/loader.md) 将其导入到 TiDB 里面进行恢复。
 
-> **注意**：虽然 TiDB 也支持使用 MySQL 官方的 `mysqldump` 工具来进行数据的备份恢复工作，但相比于 `mydumper` / `loader`，性能会慢很多，大量数据的备份恢复会花费很多时间，这里我们并不推荐。
+> **注意**：必须使用 Enterprise Tools package 的 `mydumper`，不要使用你的操作系统的 package manager 提供的 `mydumper`。`mydumper` 的上游版本并不能正确对 TiDB 进行处理 ([#155](https://github.com/maxbube/mydumper/pull/155))。由于使用 `mydumper` 进行数据备份和恢复都要耗费许多时间，这里也并不推荐。
```
Comment (kennytm) [inline op-guide/backup-restore.md]: "package" → 包
"package manager" → (软件)包管理工具

"Enterprise Tools" seems to be named as "企业版工具集" here (from what I've read from the Syncer docs).

> 由于使用 `mydumper` 进行数据备份和恢复都要耗费许多时间，这里也并不推荐。

"mydumper" should be `mysqldump` 🙃

File: tools/mydumper.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+title: mydumper 使用文档 
+summary: 使用 mydumper 从 TiDB 导出数据。
+category: tools
+---
+
+# mydumper 使用文档
+
+## mydumper 简介
+
+`mydumper` 是 [mydumper](https://github.com/maxbube/mydumper) 的 fork 项目，并添加了一些针对 TiDB 的功能。推荐使用此工具对 TiDB 进行逻辑备份。
```
Comment (kennytm) [inline tools/mydumper.md]: fork → 分支

File: tools/mydumper.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+[下载 Binary](http://download.pingcap.org/tidb-enterprise-tools-latest-linux-amd64.tar.gz)。
+
+## 相比于普通的 mydumper，此工具有哪些改进之处?
+
++ 使用 `tidb_snapshot` 而非 `FLUSH TABLES WITH READ LOCK` 提供备份一致性 
+
++ `INSERT` 语句中包含隐藏的 `_tidb_rowid` 列
+
++ 允许 `tidb_snapshot` 为 [configurable](../op-guide/history-read.md#how-tidb-reads-data-from-history-versions) （即备份之前的数据）
```
Comment (kennytm) [inline tools/mydumper.md]: configurable = "可设置的", so I'd translate it as:

```
+ [允许设置](......) `tidb_snapshot` 的值 (即是可备份不同时间点的数据)
```

BTW the anchor `#how-tidb-reads-data-from-history-versions` probably needs to be updated to match the Chinese title.

File: tools/mydumper.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+### 新添参数
+
+```
+  -z, --tidb-snapshot: Set the tidb_snapshot to be used for the backup.
+                       Default: NOW()-INTERVAL 1 SECOND.
+                       Accepts either a TSO or valid datetime.  For example: -z "2016-10-08 16:45:26"
```
Comment (kennytm) [inline tools/mydumper.md]: This also needs to be translated as well I believe.

File: tools/mydumper.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+./bin/mydumper -h 127.0.0.1 -u root -P 4000
+```
+
+## FAQ
+
+### PingCAP 的 mydumper 的源码是否可获取？
+
+PingCAP 的 mydumper 源码 [位于 GitHub](https://github.com/pingcap/mydumper)。
+
+### 是否计划未来让 PingCAP 的 mydumper 对上游的 mydumper 生效？
```
Comment (kennytm) [inline tools/mydumper.md]: I think it would be more natural to write this like

> 未来是否计划让 PingCAP 对 mydumper 的改动合并到上游？

**Comment by CaitinChen** [comment]
@kennytm Thanks a lot for your sharp eyes! 
@lilin90 PTAL

File: tools/mydumper.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+## FAQ
+
+### PingCAP 的 mydumper 的源码是否可获取？
+
+PingCAP 的 mydumper 源码 [位于 GitHub](https://github.com/pingcap/mydumper)。
+
+### 未来是否计划让 PingCAP 对 mydumper 的改动合并到上游？
+
+是的，PingCAP 团队计划让此 mydumper 对上游的 mydumper 生效。参见 [PR #155](https://github.com/maxbube/mydumper/pull/155).
```
Comment (lilin90) [inline tools/mydumper.md]: 让此 mydumper 对上游的 mydumper 生效 -> 将对 mydumper 的改动合并到上游

File: tools/mydumper.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+## mydumper 简介
+
+`mydumper` 是 [mydumper](https://github.com/maxbube/mydumper) 的 fork 项目，并添加了一些针对 TiDB 的功能。推荐使用此工具对 TiDB 进行逻辑备份。
+
+[下载 Binary](http://download.pingcap.org/tidb-enterprise-tools-latest-linux-amd64.tar.gz)。
+
+## 相比于普通的 mydumper，此工具有哪些改进之处?
+
++ 使用 `tidb_snapshot` 而非 `FLUSH TABLES WITH READ LOCK` 提供备份一致性 
+
```
Comment (lilin90) [inline tools/mydumper.md]: For such a list without sublist or detailed explanation, the blank line between each item is unnecessary. You can just remove it.

**Comment by CaitinChen** [comment]
@lilin90 Thanks. PTAL again.

File: tools/mydumper.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
++ 使用 `tidb_snapshot` 而非 `FLUSH TABLES WITH READ LOCK` 提供备份一致性 
++ `INSERT` 语句中包含隐藏的 `_tidb_rowid` 列
++ [允许设置](../op-guide/history-read.md#操作流程) `tidb_snapshot` 的值（即可备份不同时间点的数据）  
+
+### 新添参数
+
+```bash
+  -z, --tidb-snapshot: 设置 tidb_snapshot 用于备份
+                       默认值：NOW()-INTERVAL 1 SECOND
```
[the remark this answers is not in the document]
Comment (morgo) [reply tools/mydumper.md]: I've created https://github.com/pingcap/docs/pull/693 to handle the defaults change.

**Pull request merged**